app/components/steps_component.rb

```diff
@@ -0,0 +1,25 @@
+class StepsComponent < GovukComponent::Base
+  attr_reader :title
+
+  def initialize(title: nil, classes: [], html_attributes: {})
+    super(classes: classes, html_attributes: html_attributes)
+
+    @title = title
+  end
+
+  renders_many :steps, lambda { |label:, current: false, completed: false|
+    tag.li(class: "steps-component__step#{step_class(current, completed)}") { tag.h3(class: "govuk-heading-s") { label } }
+  }
+
+  def step_class(current, completed)
+    return " steps-component__step--current" if current
+
+    return " steps-component__step--completed" if completed
+  end
+
+  private
+
+  def default_classes
+    %w[steps-component govuk-!-margin-top-6]
```

Comment:
Any reason this margin shouldn't be applied by `.steps-component`?

Reply:
Nope, changed it now. I must've done this when I was fiddling with styling.